client/state/jetpack-settings/modules/actions.js

```diff
@@ -62,3 +70,28 @@ export const deactivateModule = ( siteId, moduleSlug ) => {
 			} );
 	};
 };
+
+export const fetchModuleList = ( siteId ) => {
```

Comment:
This looks great, I'm just suggesting the implementation with the 4 actions I've mentioned above.

```
export const fetchModuleList = ( siteId ) => {
	return ( dispatch ) => {
		dispatch( {
			type: JETPACK_MODULES_REQUEST,
			siteId
		} );

		return wp.undocumented().jetpackModules( siteId )
			.then( ( data ) => {
				const modules = keyBy( data.modules, 'id' );

				dispatch( {
					type: JETPACK_MODULES_RECEIVE,
					siteId,
					modules
				} );

				dispatch( {
					type: JETPACK_MODULES_REQUEST_SUCCESS,
					siteId
				} );
			} ).catch( error => {
				dispatch( {
					type: JETPACK_MODULES_REQUEST_FAILURE,
					siteId,
					error: error.message
				} );
			} );
	};
};
```